### PR TITLE
Implemented MMR

### PR DESCRIPTION
**Pull request by ofermend**
- Minor fix to word order in progress report
- Support for MMR in reranking.

### CODE REVIEW

File: src/contexts/ConfigurationContext.tsx [hunk 1]
```diff
@@ -250,7 +251,8 @@ export const ConfigContextProvider = ({ children }: Props) => {
   const [rerank, setRerank] = useState<Rerank>({
     isEnabled: false,
     numResults: 100,
-    id: 272725717
+    id: 272725717,
```
Comment (cjcenizal) [inline src/contexts/ConfigurationContext.tsx:254]: I think the diversity reranker ID is 272725718.

File: src/views/search/progressReport/ProgressReport.tsx [hunk 2]
```diff
@@ -105,7 +105,7 @@ export const ProgressReport = ({ isSearching, isSummarizing }: Props) => {
               <>
                 <VuiText>
                   <p>
-                    Retrieved search {numSearchResults} results in{" "}
+                    Retrieved {numSearchResults} search results in{" "}
```
Comment (cjcenizal) [inline src/views/search/progressReport/ProgressReport.tsx:108]: 😅

File: src/contexts/sendSearchRequest.ts [hunk 3]
```diff
@@ -84,6 +86,9 @@ export const sendSearchRequest = async ({
           ? {
               reranking_config: {
                 reranker_id: rerankerId,
+                mmrConfig: {
+                  diversityBias: rerankDiversityBias,
```
Comment (cjcenizal) [inline src/contexts/sendSearchRequest.ts:90]: Why are `mmrConfig` and `diversityBias` camel-case whereas the other properties are snake-case?

**Review by cjcenizal — COMMENTED** [review]
Looked things over and had a couple comments. Didn't test locally.

File: src/contexts/ConfigurationContext.tsx (hunk 1 above)
Comment (ofermend) [reply src/contexts/ConfigurationContext.tsx:254]: Yes this is the normal reranker. Do you think we should make the MMR the new default if rerank=true?

File: src/contexts/sendSearchRequest.ts (hunk 3 above)
Comment (ofermend) [reply src/contexts/sendSearchRequest.ts:90]: good point. I believe both work and that's why I didn't catch up, but will make consistent for the code to look better.

File: src/contexts/ConfigurationContext.tsx (hunk 1 above)
Comment (ofermend) [reply src/contexts/ConfigurationContext.tsx:254]: Ok, how about this:
1. We have two separate options
A) rerank: true/false and rerank_num_results - this enables the current reranker (which per Amin is not really very good anyways)
B) mmr: true/false with mmr_num_results and mmr_diversity_bias specific.
that way people don't have to remember the reranker_ID and just choose A or B. If they specify both we will just turn on MMR and have a log message displayed.
wdyt?

Comment (cjcenizal) [reply src/contexts/ConfigurationContext.tsx:254]: Sounds good!

File: src/contexts/sendSearchRequest.ts [hunk 4]
```diff
@@ -86,9 +86,12 @@ export const sendSearchRequest = async ({
           ? {
               reranking_config: {
                 reranker_id: rerankerId,
-                mmr_config: {
-                  diversity_bias: rerankDiversityBias,
-                }
+                ...(rerankerId === 272725718 ? {
```
Comment (cjcenizal) [inline src/contexts/sendSearchRequest.ts:89]: Perhaps extract `272725718` into a variable called `DIVERSITY_RERANKER_ID` that can be imported and used anywhere. That way we have less risk of a typo or other mistake messing this up someday.

**Review by cjcenizal — APPROVED** [review]
Code LGTM! Had one non-blocking suggestion. Didn't test locally.

File: src/contexts/sendSearchRequest.ts (hunk 4 above)
Comment (ofermend) [reply src/contexts/sendSearchRequest.ts]: Great suggestion. Done!

**Pull request merged**